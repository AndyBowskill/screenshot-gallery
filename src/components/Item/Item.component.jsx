import React from 'react';
import './Item.style.css'

const Item = (props) => {
  return (
    <div className='item'>
        <img
          className=''
          alt={props.item.url}
          src={props.item.screenshot}
          width='300px'
          height='169px'
        />
        <p>{props.item.url}</p>
    </div>
  );
};

export default Item;