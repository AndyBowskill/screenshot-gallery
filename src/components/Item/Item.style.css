.item {
    display: flex;
    flex-direction: column;
    width: 300px;
    border-radius: 5px;
    padding: 20px;
    cursor: pointer;
    -moz-osx-font-smoothing: grayscale;
    backface-visibility: hidden;
    transition: transform 0.25s ease-out;
    border: 1px solid #0074D9;
  }
  
  .item:hover {
    transform: scale(1.1);
  }