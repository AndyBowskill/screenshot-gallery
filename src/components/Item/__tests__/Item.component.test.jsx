import React from 'react';
import renderer from 'react-test-renderer';
import Item from '../Item.component';

const item = {
  url: 'www.amazon.co.uk',
  screenshot: '',
};

const onDeleteButtonClick = jest.fn();

test('Item component displays correctly.', () => {
  const component = renderer.create(<Item item={item} onDeleteButtonClick={onDeleteButtonClick} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
