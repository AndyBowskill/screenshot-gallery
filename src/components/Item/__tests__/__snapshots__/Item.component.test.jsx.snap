// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component displays correctly. 1`] = `
<article
  className="item"
>
  <img
    alt="www.amazon.co.uk"
    height="300px"
    src=""
    width="400px"
  />
  <p>
    www.amazon.co.uk
  </p>
</article>
`;
